### PR TITLE
Adds fhir_import

### PR DESCRIPTION
**Pull request by bwalsh**
This PR adds fhir_import job and cleans up docker image

### CODE REVIEW

File: docs/file-upload.txt
Comment (lbeckman314) [inline docs/file-upload.txt:1]: Is this Mermaid or D3? Very cool.

File: etl-job/fhir_import.py [hunk 1]
```diff
@@ -12,17 +13,21 @@
...
+    return os.environ.get('ACCESS_TOKEN', None)
 
 
 def _auth(access_token: str) -> Gen3Auth:
     """Authenticate using ACCESS_TOKEN"""
     # print("[out] authorizing...")
-    return Gen3Auth(refresh_file=f"accesstoken:///{access_token}")
+    if access_token:
+        # use access token from environment (set by sower)
+        return Gen3Auth(refresh_file=f"accesstoken:///{access_token}")
```
Comment (lbeckman314) [inline etl-job/fhir_import.py:24]: Does Sower pass the access_token to Fence at all? Haven't seen it or the "accesstoken:///" syntax before.

**Review by lbeckman314 — APPROVED** [review]
LGTM! Added a couple questions to fill in my knowledge gaps.

Playing around with creating a [developer wiki here](https://github.com/ACED-IDP/gen3-helm/wiki) and can the Sower doc (aced-jobs.md) there unless it would be better to keep it within the established docs we have going in `Helm/docs`.

**Docker builds successfully:**
```
➜  aced_etl_pod git:(feature/improve-setup) docker build etl-job
[+] Building 129.6s (29/29) FINISHED
 => [internal] load build definition from Dockerfile
 => ...
 => [24/24] RUN curl  https://raw.githubusercontent.com/bmeg/iceberg-schema-tools/main/config.yaml -o config.yaml
 => exporting to image                                                                                                                                                                                       2.0s
 => => exporting layers                                                                                                                                                                                      2.0s
 => => writing image sha256:568...
```

File: docs/file-upload.txt
Comment (bwalsh) [reply docs/file-upload.txt:1]: https://sequencediagram.org/

File: etl-job/fhir_import.py (hunk 1 above)
Comment (bwalsh) [reply etl-job/fhir_import.py]: https://github.com/uc-cdis/gen3sdk-python/blob/master/gen3/auth.py#L136

File: etl-job/fhir_import_export.py [hunk 2]
```diff
@@ -159,10 +228,27 @@ def _main():
...
+        _put(input_data, output, program, project, user)
+    elif method.lower() == 'get':
+        # read fhir store, write to bucket
+        object_id = _get(input_data, output, program, project, user)
+        output['object_id'] = object_id
+    else:
+        raise Exception(f"unknown method {method}")
+
+    # note, only the last output (a line in stdout with `[out]` prefix) is returned to the caller
+    print(f"[out] {json.dumps(output, separators=(',', ':'))}")
```
Comment (lbeckman314) [inline etl-job/fhir_import_export.py:244]: Thanks for finding and recording this!

File: etl-job/fhir_import_export.py [hunk 3]
```diff
@@ -141,6 +188,28 @@ def _load_all(study, project_id, output) -> bool:
...
+
+    study_path = f"studies/{project}"
+    project_id = f"{program}-{project}"
+
+    logs = fhir_get(f"{program}-{project}", study_path, DEFAULT_ELASTIC)
+    output['logs'].extend(logs)
+
+    # zip and upload the exported files to bucket
+    config = Config()
+    cp_result = cp(config=config, from_=study_path, project_id=project_id, ignore_state=False)
```
Comment (lbeckman314) [inline etl-job/fhir_import_export.py:206]: Looks good! Does `ignore_state=False` mean it overwrites existing data or keeps it in place?

https://github.com/ACED-IDP/gen3_util/blob/main/gen3_util/meta/uploader.py#L31

File: etl-job/README.md [hunk 4]
```diff
@@ -15,3 +15,46 @@ docker build -t quay.io/ohsu-comp-bio/aced-etl-job . --platform linux/amd64
...
+docker buildx imagetools inspect quay.io/ohsu-comp-bio/aced-etl-job:testing | grep Platform
+  Platform:    linux/amd64
+  Platform:    linux/arm64
+  Platform:    unknown/unknown
+  Platform:    unknown/unknown
+```
+
+View on quay.io
+
+<img width="1134" alt="image" src="https://github.com/ACED-IDP/data_model/assets/47808/5102dbae-71e2-473f-95de-7f43840c034e">
```
Comment (lbeckman314) [inline etl-job/README.md:40]: I'll have to remember this trick for images 👍

**Review by lbeckman314 — APPROVED** [review]

**Review by matthewpeterkort — APPROVED** [review]
Seems like many files got simplified here and the FHIR metadata export feature also got added. Will test the new 'test' image and report back from gen3_util repo

File: etl-job/fhir_import_export.py (hunk 3 above)
Comment (bwalsh) [reply etl-job/fhir_import_export.py:206]: over writes existing data.  will improve docs in next PR

**Pull request merged**